core/src/test/java/org/elasticsearch/index/seqno/CheckpointsIT.java

```diff
@@ -60,17 +60,17 @@ public void testCheckpointsAdvance() throws Exception {
                 logger.debug("seq_no stats for {}: {}", shardStats.getShardRouting(),
                     XContentHelper.toString(shardStats.getSeqNoStats(),
                         new ToXContent.MapParams(Collections.singletonMap("pretty", "false"))));
-                final Matcher<Long> localCheckpointRule;
+                final Matcher<Long> globalCheckpointRule;
                 if (shardStats.getShardRouting().primary()) {
-                    localCheckpointRule = equalTo(numDocs - 1L);
+                    globalCheckpointRule = equalTo(numDocs - 1L);
                 } else {
-                    // nocommit:  recovery doesn't transfer local checkpoints yet (we don't persist them in lucene).
-                    localCheckpointRule = anyOf(equalTo(numDocs - 1L), equalTo(SequenceNumbersService.NO_OPS_PERFORMED));
+                    // nocommit: recovery doesn't transfer global checkpoints yet (we don't persist them in lucene).
+                    globalCheckpointRule = anyOf(equalTo(numDocs - 1L), equalTo(SequenceNumbersService.UNASSIGNED_SEQ_NO));
                 }
                 assertThat(shardStats.getShardRouting() + " local checkpoint mismatch",
-                    shardStats.getSeqNoStats().getLocalCheckpoint(), localCheckpointRule);
+                    shardStats.getSeqNoStats().getLocalCheckpoint(), equalTo(numDocs - 1L));
```

Comment:
I think we still need the old leniency for localCheckpoints as well? if recovery completes after all docs were indexed, we will not have a local checkpoint on the replica?